### PR TITLE
Add option to use simulation time for rviz trajectory display

### PR DESCRIPTION
**Pull request by Martin-Oehler**
### Description

The Rviz MoveIt Trajectory Display uses wall time by default for state display timing. This leads to unexpected results when working with simulation time (e.g. gazebo, bag files) with less than real time. The simulated robot speed does not match the trajectory speed visualization since they follow different clocks.

This PR adds  an additional option to use simulation time instead of wall time named "Use Sim Time" (see screenshot).

image

image

### Checklist
- [x] **Required by CI**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
- [ ] Extend the tutorials / documentation [reference](http://moveit.ros.org/documentation/contributing/)
- [ ] Document API changes relevant to the user in the [MIGRATION.md](https://github.com/ros-planning/moveit/blob/master/MIGRATION.md) notes
- [ ] Create tests, which fail without this PR [reference](https://ros-planning.github.io/moveit_tutorials/doc/tests/tests_tutorial.html)
- [x] Include a screenshot if changing a GUI
- [ ] While waiting for someone to review your request, please help review [another open pull request](https://github.com/ros-planning/moveit/pulls) to support the maintainers

[//]: # "You can expect a response from a maintainer within 7 days. If you haven't heard anything by then, feel free to ping the thread. Thank you!"

### CODE REVIEW

File: moveit_ros/visualization/rviz_plugin_render_tools/src/trajectory_visualization.cpp [hunk 1]
```diff
@@ -102,6 +102,11 @@ TrajectoryVisualization::TrajectoryVisualization(rviz::Property* widget, rviz::D
   state_display_time_property_->addOptionStd("0.1s");
   state_display_time_property_->addOptionStd("0.5s");
 
+  use_sim_time_property_ = new rviz::BoolProperty("Use Sim Time", false,
+                                                  "Indicates wether simulation time or wall-time is "
```
Comment (v4hn) [inline moveit_ros/visualization/rviz_plugin_render_tools/src/trajectory_visualization.cpp:106]:
```suggestion
                                                  "Indicates whether simulation time or wall-time is "
```

**Review by v4hn — APPROVED** [review]
Looks good to me (aside from the typo) and CI passes.

**Review by tylerjw — APPROVED** [review]
lgtm except the same typo @v4hn pointed out.

File: moveit_ros/visualization/rviz_plugin_render_tools/src/trajectory_visualization.cpp [hunk 2]
```diff
@@ -219,6 +224,10 @@ void TrajectoryVisualization::changedLoopDisplay()
     trajectory_slider_panel_->pauseButton(false);
 }
 
+void TrajectoryVisualization::changedUseSimTime()
+{
+}
+
```
Comment (rhaschke) [inline moveit_ros/visualization/rviz_plugin_render_tools/src/trajectory_visualization.cpp:230]: Remove.

**Review by rhaschke — COMMENTED** [review]
Generally looks good to me as well. Please consider the suggested cleanup.

File: moveit_ros/visualization/rviz_plugin_render_tools/src/trajectory_visualization.cpp (hunk 1 above)
Comment (Martin-Oehler) [reply moveit_ros/visualization/rviz_plugin_render_tools/src/trajectory_visualization.cpp]: Fixed

File: moveit_ros/visualization/rviz_plugin_render_tools/src/trajectory_visualization.cpp (hunk 2 above)
Comment (Martin-Oehler) [reply moveit_ros/visualization/rviz_plugin_render_tools/src/trajectory_visualization.cpp]: Done

**Comment by Martin-Oehler** [comment]
I implemented your suggestions and re-based because it would no longer build with the new srdfdom version.

**Review by rhaschke — APPROVED** [review]

**Review by vatanaksoytezer — APPROVED** [review]
Also looks good to me

**Pull request merged**